Barotrauma/BarotraumaClient/ClientSource/Sounds/SoundPlayer.cs

```diff
@@ -655,6 +657,19 @@ private static void UpdateMusic(float deltaTime)
 
                 foreach (BackgroundMusic intensityMusic in suitableIntensityMusic)
                 {
+                    //if the current maintrack is meant to mute all intensity music, loop through all the intensity channels and set them to null, then continue
+                    if ((targetMusic[mainTrackIndex] != null) && (targetMusic[mainTrackIndex].MuteIntensityMusic))
+                    {
+                        for (int i = intensityTrackStartIndex; i < MaxMusicChannels; i++)
+                        {
+                            if (targetMusic[i] != null)
+                            {
+                                targetMusic[i] = null;
+                                break;
+                            }
+                        }
+                        continue;
+                    }
```

Comment:
I think this code block is quite confusing.
- Why do we need to do this for each intensity music track inside the `foreach` loop, when it stops all of the intensity tracks?
- Why does it break from the for loop as soon as one non-null target music is found (leaving the rest playing?)
- Why continue, as opposed to break?
- What's the point of first finding the suitable intensity music, and then later muting all of the intensity layers if it's not intended to play?

To me it'd seem much clearer if the method first checked if the intensity music is supposed to play, and if not, mute all the intensity tracks and not run the rest of the intensity music logic.

Reply:
I ended up removing my implementation of "MuteIntensityMusic" altogether because it seems to have already been added to the game at some point during this PR's existence in the form of "MuteIntensityTracks" which does exactly the same thing and is already coded better, so I guess problem solved